src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -69,10 +69,7 @@ public function copy($originFile, $targetFile, $override = false)
                 throw new IOException(sprintf('Failed to copy "%s" to "%s".', $originFile, $targetFile), 0, null, $originFile);
             }
 
-            if (is_executable($originFile)) {
-                // User Executable | Group Executable | Other Executable
-                chmod($targetFile, fileperms($targetFile) | 0x0040 | 0x0008 | 0x0001);
-            }
+            @chmod($targetFile, fileperms($targetFile) | (fileperms($originFile) & 0111));
```

Comment:
What about adding a comment above explaining what it does and why we only keep the executable bit?